src/DataTable/CardView.jsx

```diff
@@ -37,21 +38,54 @@ function CardItem({
   return <CardComponent {...row} />;
 }
 
+function DefaultSkeletonCardComponent() {
+  return (
+    <Card isLoading data-testid="default-skeleton-card-component">
+      <Card.ImageCap logoSkeleton />
+      <Card.Section className="pgn__data-table-card-view-default-skeleton-card-section" />
+      <Card.Footer />
+    </Card>
+  );
+}
+
+export const DEFAULT_SKELETON_CARD_COUNT = 8;
+
 function CardView({
-  columnSizes, CardComponent, className, selectionPlacement,
+  columnSizes,
+  CardComponent,
+  className,
+  selectionPlacement,
+  SkeletonCardComponent = DefaultSkeletonCardComponent,
+  skeletonCardCount,
 }) {
   const {
     getTableProps, prepareRow, displayRows,
   } = useRows();
-  const { isSelectable, manualSelectColumn } = useContext(DataTableContext);
+  const {
+    isSelectable,
+    manualSelectColumn,
+    isLoading,
+  } = useContext(DataTableContext);
+
   // use the same component for card selection that is used for row selection
   // otherwise view switching might break if row selection uses component that supports backend filtering / sorting
-  const selectionComponent = manualSelectColumn?.Cell || selectColumn.Cell;
+  const SelectionComponent = manualSelectColumn?.Cell || selectColumn.Cell;
```

Comment:
[curious]
Would rewriting the logic as `const SelectionComponent = manualSelectColumn?.Cell || selectColumn?.Cell;` prevent non-loading page instances if `selectColumn` happens to be undefined?

Reply:
@brobro10000 The `selectColumn` variable (defined in `getVisibleColumns`) is a constant, so it'll always exist :)